bulls_and_cows/lib/api.ex

```diff
@@ -0,0 +1,34 @@
+defmodule BullsAndCows.API do
+  alias BullsAndCows
+  use GenServer
+
+  # Client
+
+  @doc """
+    Make number to guess to be picked, start the game
+  """
+  def start_link() do
+    GenServer.start_link(__MODULE__, nil)
+  end
+
+  @doc """
+    Guess the number
+  """
+  def make_turn(pid, answer) do
+    GenServer.call(pid, {:make_turn, answer})
```

Comment:
With a named server, this might be simplified to
```suggestion
  def make_turn(pid \\ __MODULE__, answer) do
    GenServer.call(pid, {:make_turn, answer})
```

The drawback is you cannot now spawn _two_ of those, because names are to be unique across the system. But here it should not be an issue.